test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,156 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+public class AddCommandTest {
+    private static final List<ReadOnlyPerson> LAST_SHOWN_LIST = Collections.emptyList();
+    private static final Set<String> NO_TAGS = Collections.emptySet();
+
+    @Test
+    public void addCommand_invalidName_throws() {
+        final String[] invalidNames = { "", " ", "[]\\[;]" };
+        for (String name : invalidNames) {
+            assertConstructingInvalidAddCmdThrows(name, Phone.EXAMPLE, false, Email.EXAMPLE, false,
+                    Address.EXAMPLE, false, NO_TAGS);
```

Comment:
Seems we always use `false` as the test input. It's better to mix it up and use `true` in some places. Increasing the variety of test inputs can increase the chance of finding unexpected bugs.